src/bio/bio_monitor.c

```diff
@@ -33,6 +33,12 @@
 /* Used to preallocate buffer to query error log pages from SPDK health info */
 #define DAOS_MAX_ERROR_LOG_PAGES 256
 
+struct bio_dev_state *
+bio_get_dev_state(struct bio_xs_context *xs)
```

Comment:
I think two ways to copy out the dev state exclusively accessed by device owner xstream:
1. Always call this bio_get_dev_state() in the owner xstream context and return a copy of bio_dev_state. That probably require BIO to export an API to get owner xstream id.
2. In this bio_get_dev_state(), call spdk_thread_send_msg() to the owner xstream, and copy out the bio_dev_state in the owner xstream context.

What do you think?

Reply:
right, I think the second option is better to make sure that it is always being called from the device owner xstream and not rely on the calling function to call it from the correct xstream. that way we can take advantage of owner_xstream() and spdk_thread_send_msg() functions